### ISSUE TITLE
Accessors of type java.net.URI fields being flagged as EI

### ISSUE THREAD

**Issue by candrews** [spotbugs/spotbugs#1653]
Spotbugs 4.4.0 is flagging getters of type `java.net.URI` fields with EI_EXPOSE_REP and setters with EI_EXPOSE_REP2 indicating that these methods expose mutable application state. However [`java.net.URI` is immutable](https://docs.oracle.com/javase/8/docs/api/java/net/URI.html), so I believe this finding is in error.

Here's a minimal example of the issue.
```java
import java.net.URI;

public class UriBugTest {
    public static final java.net.URI DEFAULT_URI= URI.create("https://github.com");
    private java.net.URI uri = DEFAULT_URI;

    public URI getUri() {
        // flagged EI_EXPOSE_REP
        return this.uri;
    }

    public void setUri(URI value) {
        // flagged EI_EXPOSE_REP2
        this.uri = value;
    }
}
```

This issue is related to https://github.com/spotbugs/spotbugs/issues/1634
